### ISSUE TITLE
PRMirror writing difference info in file

### ISSUE THREAD

**Issue by LetterJay** [HippieStation/PRMirror#12]
As seen below.

![Image](https://i.imgur.com/CbGdzQy.jpg)

**Comment by j-mie** [comment]
@LetterJay come on dude, give me atleast a link to the PR
logs too please.

**Comment by LetterJay** [comment]
Here's one of the more recent ones.
https://github.com/Citadel-Station-13/Citadel-Station-13/pull/2704/files (mapping.dm)

Log: https://pastebin.com/yEUQ4z3c

Unless this is intended. I've noticed less .rej files. Instead it seems to put the conflicts in the file itself to be directly edited.

**Comment by Partheo** [comment]
Sorry to necro an old issue, but I've been having this too. Just set up the bot this week. Here's exactly what I've been getting.

---

Here's an example PR: https://github.com/outer-haven/outer-haven/pull/31
Notice how RepoCat, the mirror bot, makes two commits- the first is the actual update from TG, and the second is a seemingly empty commit with no change. PRs like this always actually mirror the TG PR without the issue in the OP.
![image](https://user-images.githubusercontent.com/10659549/33301470-e69118e4-d3c3-11e7-8c9b-0027443523b2.png)

---

Another example: https://github.com/outer-haven/outer-haven/pull/30
Here, RepoCat only makes a single commit. It's the update from TG, plus the diff markers like OP mentioned. I have to go in myself and make a commit removing the top diff and markers manually.
![image](https://user-images.githubusercontent.com/10659549/33301515-18cc64da-d3c4-11e7-904f-ffa21a2aa9f9.png)
![image](https://user-images.githubusercontent.com/10659549/33301549-40153fb2-d3c4-11e7-9424-5ccdcb0ae766.png)

---

So far, about half of the 30 Pull Requests made by the bot on my repo contain this error. I don't know what causes it, I've just noticed that the issue is never present when RepoCat makes the second empty commit, and is always present when the commit is not there.
Additionally, it only seems to affect 1-2 files in the PR. A PR with this issue that changes 10 DM files might only have a couple files with the markers in them.
Here's the repo if you want to see yourself: https://github.com/outer-haven/outer-haven
I added the FUCK label to every mirror affected by this bug.

**Comment by ghost** [comment]
This is a non-issue.

This happens because the script in the `ToolPath` was unable to properly merge the upstream patchfile into the branch, this often happens because your downstream is heavily out of date.

tl;dr you have merge conflicts

**Comment by Partheo** [comment]
I don't believe this. Our bot was up less than 12 hours after the fork was made. I don't see this as being "heavily" out of date. Given the few hours, I would find it acceptable to see this a few times, but on half the files? No. Besides, it always seems to only affect a single file per bugged PR.

Additionally, what's the deal with the second commit I'm getting? Why does it fix things, and why doesn't Citadel need it.

**Comment by praisenarsie** [comment]
here's a wild theory, you didn't fork tgstation properly, as in you didn't click the FUCKING FORK BUTTON.

how about that for a theory, eh?

**Comment by ghost** [comment]
> Additionally, what's the deal with the second commit I'm getting? Why does it fix things, and why doesn't Citadel need it.

The second commit is a merge commit, it happens when the patchfile could not be automatically merged, and thus needs a merge commit.
